src/api/restapi.js

```diff
@@ -184,30 +184,31 @@ const useRESTRoutes = (app) => {
 
   app.post(endpoint + "/Messages", checkAuth, (req, res) => {
     // typecheck and validate parameters
-    let { message, acknowledge = false, security_group = false, repeat_num = false, freq_num = false, ttl = '', bor = '' } = req.body
-
+    let { message, acknowledge = false, users, security_group = false, repeat_num = false, freq_num = false, ttl = '', bor = '' } = req.body
 
     var userList = [];
     for (var i in users) {
       userList.push(users[i].name);
     }
 
-    if (userList.length < 1) return res.send("Users missing from request.");
-
     // validate arguments, append message.
+    if (userList.length < 1) return res.send("Users missing from request.");
     if (!message) return res.send("Broadcast message missing from request.");
 
-    newBroadcast.setTTL(ttl)
-    newBroadcast.setBOR(bor)
-
     const newBroadcast = new BroadcastService()
     newBroadcast.setUsers(userList);
+    newBroadcast.setTTL(ttl)
+    newBroadcast.setBOR(bor)
 
     // let broadcast = {}
     // set user email without plus
     newBroadcast.setUserEmail(req.user.email)
     // set repeats and durations
 
+    acknowledge === true || acknowledge == 'true' ?
```

Comment:
instead of all this, only need 

if (acknowledge) {
      newBroadcast.setAckFlag(true)
    }

Reply:
Currently, this will give a double acknowledgement with the current broadcast-service

Reply:
I made the change